storage/pebble/operation/common.go

```diff
@@ -335,14 +173,19 @@ func remove(key []byte) func(*badger.Txn) error {
 // TODO: this function is unbounded – pass context.Context to this or calling functions to allow timing functions out.
 // No errors are expected during normal operation. Any errors returned by the
 // provided handleFunc will be propagated back to the caller of iterate.
-func iterate(start []byte, end []byte, iteration iterationFunc, opts ...func(*badger.IteratorOptions)) func(*badger.Txn) error {
-	return func(tx *badger.Txn) error {
+func iterate(start []byte, end []byte, iteration iterationFunc, prefetchValues bool) func(pebble.Reader) error {
+	return func(tx pebble.Reader) error {
+		// Reverse iteration is not supported by pebble
+		if bytes.Compare(start, end) > 0 {
+			return fmt.Errorf("start key must be less than or equal to end key")
+		}
 
 		// initialize the default options and comparison modifier for iteration
-		modifier := 1
-		options := badger.DefaultIteratorOptions
-		for _, apply := range opts {
-			apply(&options)
+		maxEndWithPrefix := append(end, ffBytes...)
+
+		options := pebble.IterOptions{
+			LowerBound: start,
+			UpperBound: maxEndWithPrefix,
 		}
```

Comment:
## ⚠️ ❓ 

Previously, `iterate` was executed as part of a transaction and therefore always operated on a consistent snapshot. 

I think this is no longer the case, meaning the results of the iterator might cover **incomplete records of different states**. See Pebble's documentation: https://github.com/cockroachdb/pebble/blob/1010734315f6422c1f7a4fac68140d64d3eafa30/options.go#L150-L177

I think we need to set `OnlyReadGuaranteedDurable` to `true`:
> The visible state represents a consistent point in the history of the DB.

Note that the documentation emphasizes: 
> only supported by the Reader implemented by DB

Hence, I don't think this is compatible with our Write Batch, which also implements a reader.